include/ouster_ros/os_client_base_nodelet.h

```diff
@@ -19,6 +19,10 @@ class OusterClientBase : public nodelet::Nodelet {
     virtual void onInit() override;
 
    protected:
+    bool is_arg_set(const std::string& arg) {
+        return arg.find_first_not_of(' ') != std::string::npos;
```

Comment:
for my edification :joy: what makes this approach superior to using `""` and `.empty()` like we were before

Reply:
It is not superior, it was rather needed. When you use `default=""` attribute in the **arg** definition it is still rendered as if you didn't initialize the argument and it shows this as a required value. Instead I used `default=" "` which includes a space. Hence I needed to use `arg.find_first_not_of(' ') != std::string::npos` to check if an arg has been initilaized.

Reply:
An unuset arg is left unchanged during initialized. The alternative is to either expose and use `sensor::*_UNSPEC` to the launch files which in turn would leave the config value unchanged during sensor initialization, or configure each argument with an already exposed value such as `sensor::MODE_1024x10` as the default value but then the user would need to change the value whenever they want to launch the node with different values or edit the files.

Long story short I went with the white space to indicate that the value is unchanged but I am open to suggestions, specifically the two other alternatives that I pointed out.

Reply:
Ah, I understand. So now we get the correct error instead of how it used to fail because it couldn't resolve. 
Happy with your method. Don't really want to expose `*_UNSPEC` or force users to change values every time